tests/test_license.py

```diff
@@ -1,58 +1,75 @@
-#!/usr/bin/env python3
-
-# SPDX-FileCopyrightText: 2022 Henrik Sandklef
+# SPDX-FileCopyrightText: 2022 we, the people
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-import unittest
+import pytest
 
 from flict.flictlib.return_codes import FlictError
-from flict.flictlib.return_codes import ReturnCodes
 
 from flict.flictlib.license_parser import LicenseParserFactory
 from flict.flictlib.license_parser import PrettyLicenseParser
 
 
-class TestLicenseParser(unittest.TestCase):
+@pytest.fixture(autouse=True)
+def parser():
+    one_and_only_blessed_parser = LicenseParserFactory.get_parser()
+    assert (one_and_only_blessed_parser.utils)
+    yield one_and_only_blessed_parser
+
+
+@pytest.fixture
+def parse_single_license(parser):
+    yield parser.parse_license
+
+
+@pytest.fixture
+def parse_simple(parse_single_license, lic_expr):
+    yield parse_single_license([lic_expr])['license']
 
-    def _parse_and_get_simple_license(self, parser, expr):
-        return parser.parse_license([expr])['license']['name']
-    
-    def test_license_parser_parse_license(self):
-        parser = LicenseParserFactory.get_parser()
-        
-        self.assertEqual(self._parse_and_get_simple_license(parser, "MIT"), "MIT")
-        self.assertEqual(self._parse_and_get_simple_license(parser, "X11"), "X11")
 
-        self.assertEqual(self._parse_and_get_simple_license(parser, "GPL-2.0-only WITH Classpath-exception-2.0"), "GPL-2.0-only WITH Classpath-exception-2.0")
+@pytest.mark.parametrize('lic_expr', ['MIT', "GPL-2.0-only WITH Classpath-exception-2.0"])
+def test_simple_license_result_contain_expected_keys_single_license(parse_simple, lic_expr):
+    assert isinstance(parse_simple, dict)
+    keys, vals = parse_simple.items()
+    assert all((set(keys) == {"license", "type"}, set(vals) == {lic_expr, "name"}))
+    assert parse_simple["name"] == lic_expr
 
 
-        mit_and_x11 = parser.parse_license(["MIT AND X11"])
-        self.assertEqual(mit_and_x11['license']['type'], "operator")
-        self.assertEqual(mit_and_x11['license']['name'], "AND")
+@pytest.mark.parametrize('lic_expr', ['MIT AND X11', ])
+def test_simple_license_contain_expected_keys_two_licenses(parse_simple, lic_expr):
+    assert (isinstance(parse_simple, dict))
+    p = parse_simple
+    assert p["name"] == "AND"
+    assert len(p["operands"]) == 2
+    assert p["type"] == "operator"
+    assert all((lic["type"] == "license" for lic in p["operands"]))
+    assert all((lic["name"] in ("MIT", "X11") for lic in p["operands"]))
 
-        operands = mit_and_x11['license']['operands']
 
-        self.assertTrue(operands[0]['name'] == "MIT" and operands[1]['name'] == "X11" or
-                            operands[1]['name'] == "MIT" and operands[0]['name'] == "X11")
+@pytest.mark.parametrize('lic_expr', ["This is not a list", ])
+def test_fail_on_invalid_str_license_expression(parse_simple):
+    # with pytest.raises(FlictError):  #  FIXME flict does not fail here
```

Comment:
Huch, really? That's worth a new bug ticket

Reply:
yup, it surprised me to that degeree that i didn;t trust pytest and ran
```shell
$ flict  simplify "This is not a list"
{"original": "This is not a list", "simplified": "This is not a list"}
```

Reply:
@hesa your turn

Reply:
I guess #231 fulfills this request.

Reply:
Indeed it does